// nuxt-testimonial/dist/module.mjs
import { defineNuxtModule, createResolver, addPlugin, addImports, addComponent, addServerHandler } from '@nuxt/kit';

const module = defineNuxtModule({
  meta: {
    name: "nuxt-testimonial",
    configKey: "nuxtTestimonial"
  },
  // Default configuration options of the Nuxt module
  defaults: {},
  setup(options, nuxt) {
    const { resolve } = createResolver(import.meta.url);
    addPlugin(resolve("./runtime/vue-masonry"));
    addImports({
      name: "useElementIntersection",
      as: "useElementIntersection",
      from: resolve("./runtime/composables/useElementIntersection")
    });
    addComponent({
      name: "NuxtTweet",
      filePath: resolve("./runtime/components/NuxtTweet.vue")
    });
    addComponent({
      name: "NuxtTestimonial",
      filePath: resolve("./runtime/components/NuxtTestimonial.vue")
    });
    addServerHandler({
      route: "/api/tweets/:id",
      handler: resolve("./runtime/server/api/tweets/[id].get")
    });
  }
});

export { module as default };
