// @yeger/vue-masonry-wall/dist/index.mjs
import { defineComponent as k, mergeDefaults as w, ref as p, nextTick as h, onBeforeUnmount as C, onMounted as M, watch as v, toRefs as W, openBlock as l, createElementBlock as o, normalizeStyle as u, Fragment as c, renderList as y, unref as x, renderSlot as B, createTextVNode as S, toDisplayString as $ } from "vue";
import { defaults as b, useMasonryWall as T } from "@yeger/vue-masonry-wall-core";
const z = ["data-index"], D = /* @__PURE__ */ k({
  __name: "masonry-wall",
  props: w({
    columnWidth: {},
    items: {},
    gap: {},
    rtl: { type: Boolean },
    ssrColumns: {},
    scrollContainer: {},
    minColumns: {},
    maxColumns: {},
    keyMapper: { type: Function }
  }, b),
  emits: ["redraw", "redrawSkip"],
  setup(t, { emit: s }) {
    const f = t, r = p([]), i = p(), { getColumnWidthTarget: d } = T({
      columns: r,
      emit: s,
      nextTick: h,
      onBeforeUnmount: C,
      onMounted: M,
      vue: 3,
      wall: i,
      watch: v,
      ...W(f)
    });
    return (e, F) => (l(), o("div", {
      ref_key: "wall",
      ref: i,
      class: "masonry-wall",
      style: u({ display: "flex", gap: `${e.gap}px` })
    }, [
      (l(!0), o(c, null, y(r.value, (g, n) => (l(), o("div", {
        key: n,
        class: "masonry-column",
        "data-index": n,
        style: u({
          display: "flex",
          "flex-basis": `${x(d)(n)}px`,
          "flex-direction": "column",
          "flex-grow": 1,
          gap: `${e.gap}px`,
          height: [
            "-webkit-max-content",
            "-moz-max-content",
            "max-content"
          ],
          "min-width": 0
        })
      }, [
        (l(!0), o(c, null, y(g, (a, m) => (l(), o("div", {
          key: e.keyMapper(e.items[a], n, m, a),
          class: "masonry-item"
        }, [
          B(e.$slots, "default", {
            item: e.items[a],
            column: n,
            row: m,
            index: a
          }, () => [
            S($(e.items[a]), 1)
          ])
        ]))), 128))
      ], 12, z))), 128))
    ], 4));
  }
}), L = /* @__PURE__ */ (() => {
  const t = D;
  return t.install = (s) => {
    s.component("MasonryWall", t);
  }, t;
})();
export {
  L as default
};
